server/zanata-war/src/main/java/org/zanata/adapter/JsonAdapter.java

```diff
@@ -46,4 +48,9 @@ protected void updateParamsWithDefaults(IParameters params) {
         Parameters p = (Parameters) params;
         p.fromString("extractIsolatedStrings.b=true");
     }
+
+    @Override
+    protected String getIdFor(TextUnit tu, String content, String subDocName) {
+        return HashUtil.generateHash(tu.getName().concat(content));
```

Comment:
I'm not sure that this was the right fix.  It:
- uses the key as part of the ID if it exists
- works afaik
- will not be compatible with currently uploaded translations